homeassistant/components/totalconnect/__init__.py

```diff
@@ -1 +1,51 @@
 """The totalconnect component."""
+import logging
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers import discovery
+from homeassistant.const import (CONF_PASSWORD, CONF_USERNAME)
+from total_connect_client import TotalConnectClient
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'totalconnect'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+TOTALCONNECT_PLATFORMS = ['alarm_control_panel']
+
+
+def setup(hass, config):
+    """Set up TotalConnect component."""
+    conf = config[DOMAIN]
+
+    username = conf[CONF_USERNAME]
+    password = conf[CONF_PASSWORD]
+
+    client = TotalConnectClient.TotalConnectClient(username, password)
+
+    if client.token is False:
+        return False
```

Comment:
Log an error here and say why we can't set up the component.

Reply:
added error message